openquake/baselib/parallel.py

```diff
@@ -608,55 +602,64 @@ def wait(self):
         """
         return self.reduce(self, lambda acc, res: acc + 1, 0)
 
-    def submit_all(self):
+    @property
+    def num_tasks(self):
         """
-        :returns: an IterResult object
+        The number of tasks, if known, or the empty string.
         """
         try:
-            nargs = len(self.task_args)
+            return len(self.task_args)
         except TypeError:  # generators have no len
-            nargs = ''
-        if nargs == 1:
-            [args] = self.task_args
-            add_task_no(args, 0)
+            return ''
```

Comment:
Please return -1 or something that makes explicit that you can't extract the len from a generator

Reply:
Done